File: 2022/day1/day1.lua
file = io.open("input1.txt", "r")
io.input(file)

max = {0, 0, 0}
i = 0

while true do
    local line = io.read("*line")
    if line == nil then break end

    if line == "" then
        for k=1, #max do
            if max[k] < i then
                max[k] = i
                break
            end
        end
        i = 0
    else
        i = i + tonumber(line)
    end
end

--part 1, 66306
print(max[1])

--part 2, 195292
sum = 0
for k=1, #max do
    sum = sum + max[k]
end
print(sum)

File: 2022/day1/day1.rb
def part1
  max_sum = 0
  bucket = []
  File.open("input.txt", "r").each do |line|
    line.chomp!

    if line != "" 
      bucket.push line.to_i
    else 
      max_sum = [max_sum, bucket.sum].max
      bucket = []
    end
  end

  return max_sum
end

def part2
  sum = 0
  bucket = []
  File.open("input.txt", "r").each do |line|
    line.chomp!

    if line != "" 
      sum += line.to_i
    else 
      bucket.push sum
      sum = 0
    end
  end

  # return bucket.max(3).sum
  # what is faster? 
  return bucket.sort_by{|n| -n}.take(3).sum
end


puts "Part1: #{part1}"
puts "Part2: #{part2}"